docusaurus/docs/React/guides/sdk-state-management.mdx

```diff
@@ -0,0 +1,146 @@
+---
+id: sdk-state-management
+title: SDK State Management
+---
+
+Most of the application state which is the main driver behind the UI changes of our chat SDK lives within our LLC ([`StreamChat`](https://github.com/GetStream/stream-chat-js)) - more specifically client and channel instances and is _NOT REACTIVE_ - our React SDK makes these state stores reactive by listening to the same events client does and copies it with the help of `useState` API after it has been changed.
+
+This is how the SDK state pipeline looks like behind the scenes:
+
+```txt
+client attaches WS event listeners -> React SDK attaches WS event listeners -> WS event received -> client/channel instance updates its own state -> React SDK pulls the updated state and updates it locally
+```
+
+Optimistic updates - if applicable - are handled by the React SDK.
+
+## Active Channel & Channel State
+
+Before you can access the _reactive_ channel state you'll need to make one of the available channels active either through the use of the [`ChannelList`](../components/core-components/channel-list.mdx) component which handles this for you ([default setup](../basics/getting-started.mdx#your-first-app-with-stream-chat-react)) or by passing the `channel` instance to a [`Channel`](../components/core-components/channel.mdx) component as a prop or lastly, by calling `setActiveChannel`.
```

Comment:
I find longer lists like that easier to understand when they are broken up into multiple lines, e.g.:

> The reactive channel state is accessible when the channel is marked as active. The channel becomes active when
> 1. It's selected in the `ChannelList` component; if you're using our default setup, you probably already have `ChannelList` in your application.
> 2. It's passed to the `channel` prop of the `Channel` component.
> 3. It's explicitly marked as active by calling the `setActiveChannel` method on the client instance.

Reply:
We should also link from `setActiveChannel` (if the method is documented somewhere :)

Reply:
If you decide to go this way, you can also move the examples closer to the actual cases. Then it flows nicer: here's how you mark channel is active; the active channel lives in `ChannelStateContext`; here's how you access the context.